ms_active_directory/core/ad_session.py

```diff
@@ -263,6 +263,92 @@ def _create_object(self, object_dn: str, object_classes: List[str], account_attr
                                       'and object classes {}. LDAP result: {}'.format(object_dn, object_classes,
                                                                                       result))
 
+    def create_user(self, username: str, first_name: str, last_name: str, object_location: str,
+                    **additional_user_attributes) -> ADUser:
```

Comment:
is there a reason we don't want to require specifying a password on user creation? and then handle the encoding of it as necessary, generation of user keys, etc.? it seems like it would be good to have a `ManagedUser` object (similar to the `ManagedComputer` object for computer creation) that accounts for the password and can handle generating things like kerberos keys for the user upon/after creation.

part of why I didn't add user creation yet is that it's a bit more involved than groups since users have the different auth mechanisms, encryption types, etc.

maybe it would make sense to name this `create_unmanaged_user` or `create_precreated_user` or something like that, to differentiate it from users that are ready for use?

"precreated computer account" is a somewhat common term for when you create a computer account but don't configure credentials for it because it'll be later taken over by the intended workstation that will then reset the credentials. so I'm thinking that this would be that, but for users, since no user can actually login as a user created with this function as-is

Reply:
A `ManagedUser` object would definitely make sense for some contexts, but I think creating an unmanaged user also makes sense. I like the idea of naming this `create_unmanaged_user` that way if desired, the auth mechanisms can be manually set up. In the future, adding another more general purpose `create_user` function that handles authentication configuration would be great.

Reply:
renaming this to `create_unmanaged_user` and then leaving the general behavior as-is sounds good to me :)

Reply:
Done.